### PR TITLE
Move compiler_stub to intrinsiclib

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #1354 

The patch has passed the follow build: VS/GCC/CLANG for x64 and ARM_DS2022/ARM_GNU/AARCH64_GCC for aarch64.
image

image

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: unit_test/test_size/test_size_of_spdm_requester/CMakeLists.txt [hunk 1]
```diff
@@ -27,8 +27,8 @@ SET(src_test_size_of_spdm_requester
     spdm_requester_init.c
     spdm_requester_main.c
     spdm_requester_session.c
-    compiler_stub.c
     support.c
+    ${LIBSPDM_DIR}/unit_test/test_size/intrinsiclib/compiler_stub.c
```
Comment (jyao1) [inline unit_test/test_size/test_size_of_spdm_requester/CMakeLists.txt:31]: why we need this?

File: unit_test/test_size/test_size_of_spdm_responder/CMakeLists.txt [hunk 2]
```diff
@@ -25,8 +25,8 @@ INCLUDE_DIRECTORIES(${LIBSPDM_DIR}/unit_test/test_size/test_size_of_spdm_respond
 SET(src_test_size_of_spdm_responder
     spdm_responder_init.c
     spdm_responder_main.c
-    compiler_stub.c
     support.c
+    ${LIBSPDM_DIR}/unit_test/test_size/intrinsiclib/compiler_stub.c
```
Comment (jyao1) [inline unit_test/test_size/test_size_of_spdm_responder/CMakeLists.txt:29]: why we need this?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -92,6 +92,9 @@ jobs:
             toolchain: ARM_GNU
           - crypto: openssl
             toolchain: ARM_GNU
+          - configurations: "-DLIBSPDM_ENABLE_CAPABILITY_CERT_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_CHAL_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_MEAS_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_KEY_EX_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_PSK_EX_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_SET_CERTIFICATE_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_CHUNK_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP=1 -DLIBSPDM_ENABLE_CAPABILITY_ENCAP_CAP=1"
```
Comment (jyao1) [inline .github/workflows/build.yml:95]: why we need this?

**Review by qizhangz — COMMENTED** [review]
It seems that building intrinsiclib as shared libary can resolve __stack_chk_fail link error


diff --git a/unit_test/test_size/intrinsiclib/CMakeLists.txt b/unit_test/test_size/intrinsiclib/CMakeLists.txt
index 164efecd..0705c59c 100644
--- a/unit_test/test_size/intrinsiclib/CMakeLists.txt
+++ b/unit_test/test_size/intrinsiclib/CMakeLists.txt
@@ -42,4 +42,4 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
     endif()
 endif()

-ADD_LIBRARY(intrinsiclib STATIC ${src_intrinsiclib})
+ADD_LIBRARY(intrinsiclib SHARED ${src_intrinsiclib})

**Comment by jyao1** [comment]
Have you tested AARCH64 (ARM_DS2022, ARM_GNU, ARM_GCC) ?

**Review by qizhangz — COMMENTED** [review]
Remove -flto from compile options. The build can pass with intrinsiclib as static lib.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b19c5be..38ce3014 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,7 +179,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         if(STACK_USAGE STREQUAL "ON")
             ADD_COMPILE_OPTIONS(-fstack-usage)
         else()
-            ADD_COMPILE_OPTIONS(-flto)
+            #ADD_COMPILE_OPTIONS(-flto)
         endif()
         if(GCOV STREQUAL "ON")
         ADD_COMPILE_OPTIONS(--coverage -fprofile-arcs -ftest-coverage)

**Review by qizhangz — COMMENTED** [review]
CLANG build also need to remove -flto from compile options

     elseif(TOOLCHAIN STREQUAL "CLANG")
         SET(CMAKE_C_COMPILER clang)
-        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -flto -DUSING_LTO -Werror-implicit-function-declaration)
+        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables  -DUSING_LTO -Werror-implicit-function-declaration)

**Pull request merged**